lib/engine/step/base.rb

```diff
@@ -6,7 +6,8 @@ module Engine
   module Step
     class Base
       include Passer
-      attr_accessor :acted
+      attr_accessor :acted, :inactive
```

Comment:
why do you need inactive?

Reply:
I don't want `DiscardTrain` showing up as the `current_step` when we're really on the `Track` step. If no train needs to be discarded, `Round#skip_steps` will just call `next` when looking at the `DiscardTrain` step, as `DiscardTrain#active?` returns `false`; it doesn't get `pass!` called on it.

Up where `abilities` wants to know what the current step is, without `inactive`, it can't filter out `DiscardTrain` from the list.

My first approach was to just move the `DiscardTrain` step to right before `BuyTrain`, but this is incompatible with 18CO (see comment thread at https://github.com/tobymao/18xx/pull/2870#discussion_r549552231).

Reply:
i'd rather we handle this another way instead of making the system more complex just for 18co. can you have 18co do an explicit pass or something?

Reply:
I'll see if I can find a sample 18CO game where a takeover forced a train discard and experiment with it a bit.

Reply:
Doesn't look like any such games exist yet:

```
irb(main):008:0> Game.where(title: '18CO').all.select { |game| game.actions.any? { |a| a.action['type'] == 'discard_train' } }
=> [] 
```

I'll try and make it happen in a game now, just have to do a bit of reading up on 18CO first 😄

Reply:
Got a sample hotseat game, now to experiment with code changes - https://gist.github.com/michaeljb/9eb00cb70ad0efaf2659d9a190e95f0d

Reply:
As @ryandriskel suspected, moving DiscardTrain to right before BuyTrain breaks with the above game:

```
 GameError: Step Lay Track cannot process action {"type"=>"discard_train", "entity"=>"DSNG", "entity_type"=>"corporation", "id"=>81, "train"=>"2-3"}
```

I could have `potential_blocking_step` reject the `DiscardTrain` step since it's generally unlikely to be a step where special abilities are happening. Or maybe move `potential_blocking_step` from `Round` to `Game`, then 18CO could override it to reject `DiscardTrain`, and any other game can do something similar if needed?

Reply:
Got rid of `inactive`, moved `Round#potential_blocking_step` to `Game#ability_blocking_step`, and gave 18CO a special override to ignore `DiscardTrain` in ffefe35d, running local validation now...